processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessingContextImpl.java

```diff
@@ -90,18 +95,37 @@ public DeferredCompletion deferCompletion() {
             return CompletableFuture.completedFuture(null);
         }
 
-        CompletableFuture<Void> future = new CompletableFuture<>();
-        DeferredCompletion nextCompletion = () -> future.complete(null);
-
         DecatonTask<P> task = new DecatonTask<>(
                 this.task.metadata(), taskData, this.task.taskDataBytes());
         DecatonProcessor<P> nextProcessor = downstreams.get(0);
+        final TraceHandle parentTrace = request.trace();
+        final TraceHandle traceHandle = null == parentTrace ? NoopTrace.INSTANCE
```

Comment:
why is this null-check necessary? I think in all cases a non-null trace is supplied.

Reply:
The `TracingProvider` could have returned `null`, but a general `try`/`catch` probably covers that better. Will do that.

Reply:
I see.
However, as a protocol I think we should consider `childFor`'s return to be always non-null. 
For the guard against misimpementation of driver, I think we should guard against not only NPE but also for other exceptions so `try-catch` should be the way to go.

Reply:
Yes, agreed - I've done it that way now.